src/game/shared_game_state.rs

```diff
@@ -576,6 +576,8 @@ impl SharedGameState {
         } else {
             if (locale.code == "jp" || locale.code == "en") && constants.is_base() {
                 constants.textscript.encoding = TextScriptEncoding::ShiftJIS
+            } else if (locale.code == "jp" && constants.is_cs_plus) {
+                constants.textscript.encoding = TextScriptEncoding::ShiftJIS
```

Comment:
Does not set ShiftJIS when on JP Locale with Switch. 

For future: For better readability it may be nice to add separate if statements for each pair of [default locale + platform] or just [platform]. Along with comments to explain.